bin/migrate_to_single_table.py

```diff
@@ -0,0 +1,319 @@
+        """
+        try:
+            stmt = (
+                self.session.query(
+                    IndexRecordUrlMetadata.url,
+                    IndexRecordUrlMetadata.key,
+                    IndexRecordUrlMetadata.value,
+                )
+                .filter(IndexRecordUrlMetadata.did == did)
+                .all()
+            )
+            res = {url: {key: value} for url, key, value in stmt}
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with url metadata for {did}: {e}")
+
+    def get_index_record_ace(self, did):
+        """
+        Get the index record ace for the given did and return correctly formatted value
+        """
+        try:
+            stmt = (
+                self.session.query(IndexRecordACE.ace)
+                .filter(IndexRecordACE.did == did)
+                .all()
+            )
+            res = [a.ace for a in stmt]
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with ace for did {did}: {e}")
+
+    def get_index_record_authz(self, did):
+        """
+        Get the index record authz for the given did and return the correctly formatted value
+        """
+        try:
+            stmt = (
+                self.session.query(IndexRecordAuthz.resource)
+                .filter(IndexRecordAuthz.did == did)
+                .all()
+            )
+            res = [r.resource for r in stmt]
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with authz: {e}")
+
+    def get_index_record_alias(self, did):
+        """
+        Get the index record alias for the given did and return the correctly formatted
+        """
+        try:
+            stmt = (
+                self.session.query(IndexRecordAlias.name)
+                .filter(IndexRecordAlias.did == did)
+                .all()
+            )
+            res = {}
+            for did, name in stmt:
+                if did not in res:
+                    res[did] = []
+                res[did].append(name)
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with alias: {e}")
+
+    def get_index_record_metadata(self, did):
+        """
+        Get the index record metadata for the given did and return the correctly fortmatted value
+        """
+        try:
+            stmt = (
+                self.session.query(
+                    IndexRecordMetadata.key,
+                    IndexRecordMetadata.value,
+                )
+                .filter(IndexRecordMetadata.did == did)
+                .all()
+            )
+            res = {key: value for key, value in stmt}
+            return res
+        except Exception as e:
+            self.logger.error(f"Error with alias for did {did}: {e}")
+
+    def remove_duplicate_records(self, records, error):
+        """
+        Remove duplicate records from the bulk insert records list
```

Comment:
do you know how this happens in the first place?

Reply:
I'd say we remove this then, it's probably adding a decent chunk of time iterating through all the records